app.test.js

```diff
@@ -117,3 +117,47 @@ describe('getReviewsById', () => {
   });
 
 })
+
+describe('getCommentsByReviewId', () => {
+  test('get an array of comments for a given review id with correct properties', () => {
+    return request(app)
+      .get("/api/reviews/2/comments")
+      .expect(200)
+      .then(({body}) => {
+        body.comments.forEach((comment) => {
+          expect(comment).toMatchObject({
+            comment_id: expect.any(Number),
+            votes:expect.any(Number),
+            created_at:expect.any(String),
+            author:expect.any(String),
+            body:expect.any(String),
+            review_id: 2
+        });
+        })
+    });
+  })
+  test('get an arempty array when no comments exist', () => {
+    return request(app)
+      .get("/api/reviews/1/comments")
+      .expect(200)
+      .then(({body}) => {
+        expect(body.comments.length).toBe(0);
+    });
+  })
+  test("Responds with 404 when the wrong datatype is given for reviewid", () => {
+    return request(app)
+      .get("/api/reviews/nonsense/comments")
+      .expect(400)
+      .then(({body}) => {
+        expect(body.msg).toBe('bad request!')
+      });
+  });
+  test("Responds with 400 and invalid review_id given", () => {
```

Comment:
I would argue this is actually a 404, a number is absolutely the correct kind of request but this data just does not exist and is there 'not found'